blog/api/write/router.py

```
@@ -0,0 +1,550 @@
+        new_username = new_username.strip()
+        if not new_username:
+            errors["username"] = "Username cannot be empty."
+        elif User.objects.filter(username=new_username).exclude(id=user.id).exists():
+            errors["username"] = "Username already taken."
+        else:
+            user.username = new_username
+
+    if new_password is not None:
+        if not current_password:
+            errors["current_password"] = (
+                "Current password is required to change password."
```

Ruff (0.15.9)

[error] Possible hardcoded password assigned to: "current_password"

(S105)